File: ios/Packages/SolennixNetwork/Sources/SolennixNetwork/Endpoints.swift
import Foundation

/// Centralized API endpoint definitions for all Solennix API routes.
public enum Endpoint {

    // MARK: - Auth

    public static let register = "/auth/register"
    public static let login = "/auth/login"
    public static let logout = "/auth/logout"
    public static let refresh = "/auth/refresh"
    public static let forgotPassword = "/auth/forgot-password"
    public static let resetPassword = "/auth/reset-password"
    public static let me = "/auth/me"
    public static let changePassword = "/auth/change-password"
    public static let appleAuth = "/auth/apple"
    public static let googleAuth = "/auth/google"

    // MARK: - Clients

    public static let clients = "/clients"

    public static func client(_ id: String) -> String {
        "/clients/\(id)"
    }

    // MARK: - Events

    public static let events = "/events"
    public static let upcomingEvents = "/events/upcoming"

    public static func event(_ id: String) -> String {
        "/events/\(id)"
    }

    public static func eventProducts(_ id: String) -> String {
        "/events/\(id)/products"
    }

    public static func eventExtras(_ id: String) -> String {
        "/events/\(id)/extras"
    }

    public static func eventItems(_ id: String) -> String {
        "/events/\(id)/items"
    }

    public static func eventEquipment(_ id: String) -> String {
        "/events/\(id)/equipment"
    }

    public static func eventSupplies(_ id: String) -> String {
        "/events/\(id)/supplies"
    }

    public static let equipmentConflicts = "/events/equipment/conflicts"
    public static let equipmentSuggestions = "/events/equipment/suggestions"
    public static let supplySuggestions = "/events/supplies/suggestions"

    public static func eventStaff(_ id: String) -> String {
        "/events/\(id)/staff"
    }

    /// Organizer-facing endpoint for the client-portal share link of an event
    /// (PRD/12 feature A). Used by GET (fetch active), POST (create or rotate)
    /// and DELETE (revoke) — same path, different verbs.
    public static func eventPublicLink(_ id: String) -> String {
        "/events/\(id)/public-link"
    }

    // MARK: - Staff (Personal / Colaboradores)

    public static let staff = "/staff"

    public static func staff(_ id: String) -> String {
        "/staff/\(id)"
    }

    /// Reporte de disponibilidad de staff para una fecha o rango.
    /// Consumido via `GET /api/staff/availability?date=YYYY-MM-DD` o
    /// `?start=YYYY-MM-DD&end=YYYY-MM-DD`.
    public static let staffAvailability = "/staff/availability"

    // MARK: - Staff Teams (Cuadrillas)

    public static let staffTeams = "/staff/teams"

    public static func staffTeam(_ id: String) -> String {
        "/staff/teams/\(id)"
    }

    // MARK: - Products

    public static let products = "/products"

    public static func product(_ id: String) -> String {
        "/products/\(id)"
    }

    public static func productIngredients(_ id: String) -> String {
        "/products/\(id)/ingredients"
    }

    public static let batchIngredients = "/products/ingredients/batch"

    // MARK: - Inventory

    public static let inventory = "/inventory"

    public static func inventoryItem(_ id: String) -> String {
        "/inventory/\(id)"
    }

    // MARK: - Payments

    public static let payments = "/payments"

    public static func payment(_ id: String) -> String {
        "/payments/\(id)"
    }

    // MARK: - Unavailable Dates

    public static let unavailableDates = "/unavailable-dates"

    public static func unavailableDate(_ id: String) -> String {
        "/unavailable-dates/\(id)"
    }

    // MARK: - Dashboard

    /// Aggregated KPI endpoint — returns counts + totals server-side.
    /// Use this for the dashboard header so the UI can paint immediately
    /// while the underlying list endpoints continue loading.
    public static let dashboardKpis = "/dashboard/kpis"

    /// Revenue chart endpoint — returns `[DashboardRevenuePoint]` with one
    /// entry per month. Query param `period=month|quarter|year` controls the
    /// window (year = last 12 months). The dashboard's 6-month chart takes
    /// the last 6 entries.
    public static let dashboardRevenueChart = "/dashboard/revenue-chart"

    /// Events grouped by status. Query param `scope=month|all` — the
    /// dashboard passes `month` so the status distribution stays consistent
    /// with the rest of the month-scoped KPI cards.
    public static let dashboardEventsByStatus = "/dashboard/events-by-status"

    // MARK: - Search

    public static let search = "/search"

    // MARK: - Uploads

    public static let uploadImage = "/uploads/image"
    public static let uploadPresign = "/uploads/presign"
    public static let uploadComplete = "/uploads/complete"

    // MARK: - Profile

    public static let updateProfile = "/users/me"

    // MARK: - Subscriptions

    public static let subscriptionStatus = "/subscriptions/status"

    // MARK: - Devices

    public static let registerDevice = "/devices/register"
    public static let unregisterDevice = "/devices/unregister"

    // MARK: - Event Form Links

    public static let eventFormLinks = "/event-forms"

    public static func eventFormLink(_ id: String) -> String {
        "/event-forms/\(id)"
    }

    // MARK: - Live Activities (iOS Dynamic Island remote updates)

    public static let registerLiveActivityToken = "/live-activities/register"

    public static func liveActivityByEvent(_ eventId: String) -> String {
        "/live-activities/by-event/\(eventId)"
    }
}

File: android/core/network/src/main/java/com/creapolis/solennix/core/network/Endpoints.kt
package com.creapolis.solennix.core.network

object Endpoints {
    // Auth
    const val REGISTER = "auth/register"
    const val LOGIN = "auth/login"
    const val LOGOUT = "auth/logout"
    const val REFRESH = "auth/refresh"
    const val FORGOT_PASSWORD = "auth/forgot-password"
    const val RESET_PASSWORD = "auth/reset-password"
    const val VERIFY_EMAIL_RESEND = "auth/verify-email/resend"
    const val TEAM_INVITE_ACCEPT = "auth/team-invite/accept"
    const val ME = "auth/me"
    const val CHANGE_PASSWORD = "auth/change-password"
    const val GOOGLE_AUTH = "auth/google"
    const val APPLE_AUTH = "auth/apple"

    // Clients
    const val CLIENTS = "clients"
    fun client(id: String) = "clients/$id"

    // Staff (Personal / Colaboradores)
    const val STAFF = "staff"
    const val STAFF_AVAILABILITY = "staff/availability"
    const val STAFF_MY_ASSIGNMENTS = "staff/my-assignments"
    const val STAFF_MY_TIMELINE = "staff/my-timeline"
    const val STAFF_MY_TIMELINE_READ = "staff/my-timeline/read"
    fun staff(id: String) = "staff/$id"
    fun staffInvite(id: String) = "staff/$id/invite"
    fun staffRespondAssignment(id: String) = "staff/assignments/$id/respond"
    fun eventStaff(eventId: String) = "events/$eventId/staff"

    // Staff Teams (Equipos)
    const val STAFF_TEAMS = "staff/teams"
    fun staffTeam(id: String) = "staff/teams/$id"

    // Events
    const val EVENTS = "events"
    const val UPCOMING_EVENTS = "events/upcoming"
    const val EVENTS_SEARCH = "events/search"
    const val EVENTS_ICAL = "events/ical"
    fun event(id: String) = "events/$id"
    fun eventProducts(id: String) = "events/$id/products"
    fun eventExtras(id: String) = "events/$id/extras"
    fun eventItems(id: String) = "events/$id/items"
    fun eventEquipment(id: String) = "events/$id/equipment"
    fun eventSupplies(id: String) = "events/$id/supplies"
    fun eventPhotos(id: String) = "events/$id/photos"
    fun eventPhoto(eventId: String, photoId: String) = "events/$eventId/photos/$photoId"
    fun eventPublicLink(eventId: String) = "events/$eventId/public-link"
    fun eventPdf(eventId: String, type: String) = "events/$eventId/pdf/$type"
    const val EQUIPMENT_CONFLICTS = "events/equipment/conflicts"
    const val EQUIPMENT_SUGGESTIONS = "events/equipment/suggestions"
    const val EQUIPMENT_AVAILABILITY = "events/equipment/availability"
    const val SUPPLY_SUGGESTIONS = "events/supplies/suggestions"

    // Products
    const val PRODUCTS = "products"
    fun product(id: String) = "products/$id"
    fun productIngredients(id: String) = "products/$id/ingredients"
    const val BATCH_INGREDIENTS = "products/ingredients/batch"

    // Inventory
    const val INVENTORY = "inventory"
    fun inventoryItem(id: String) = "inventory/$id"

    // Payments
    const val PAYMENTS = "payments"
    fun payment(id: String) = "payments/$id"

    // Quick Quotes
    const val QUICK_QUOTES = "quick-quotes"
    const val QUICK_QUOTES_PDF = "quick-quotes/pdf"

    // Unavailable Dates
    const val UNAVAILABLE_DATES = "unavailable-dates"
    fun unavailableDate(id: String) = "unavailable-dates/$id"

    // Search
    const val SEARCH = "search"

    // Uploads
    const val UPLOAD_IMAGE = "uploads/image"
    const val UPLOAD_PRESIGN = "uploads/presign"
    const val UPLOAD_COMPLETE = "uploads/complete"

    // Profile
    const val UPDATE_PROFILE = "users/me"

    // Subscriptions
    const val SUBSCRIPTION_STATUS = "subscriptions/status"

    // Dashboard (aggregated, server-computed)
    const val DASHBOARD_KPIS = "dashboard/kpis"
    const val DASHBOARD_REVENUE_CHART = "dashboard/revenue-chart"
    const val DASHBOARD_EVENTS_BY_STATUS = "dashboard/events-by-status"
    const val DASHBOARD_TOP_CLIENTS = "dashboard/top-clients"
    const val DASHBOARD_PRODUCT_DEMAND = "dashboard/product-demand"
    const val DASHBOARD_FORECAST = "dashboard/forecast"

    // Event Form Links
    const val EVENT_FORM_LINKS = "event-forms"
    fun eventFormLink(id: String) = "event-forms/$id"

    // Reviews (organizer inbox)
    const val REVIEWS = "reviews"
    fun reviewResponse(id: String) = "reviews/$id/response"
    fun reviewVisibility(id: String) = "reviews/$id/visibility"

    // Devices
    const val REGISTER_DEVICE = "devices/register"
    const val UNREGISTER_DEVICE = "devices/unregister"
}
